analyzers/src/SonarAnalyzer.VisualBasic/Rules/LooseFilePermissions.cs

```diff
@@ -0,0 +1,125 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.VisualBasic;
+using Microsoft.CodeAnalysis.VisualBasic.Syntax;
+using SonarAnalyzer.Common;
+using SonarAnalyzer.Extensions;
+using SonarAnalyzer.Helpers;
+using SonarAnalyzer.Rules.Hotspots;
+
+namespace SonarAnalyzer.Rules.VisualBasic
+{
+    [DiagnosticAnalyzer(LanguageNames.VisualBasic)]
+    [Rule(DiagnosticId)]
+    public sealed class LooseFilePermissions : LooseFilePermissionsBase<SyntaxKind>
+    {
+        protected override ILanguageFacade<SyntaxKind> Language => VisualBasicFacade.Instance;
+
+        public LooseFilePermissions() : this(AnalyzerConfiguration.Hotspot) { }
+
+        internal LooseFilePermissions(IAnalyzerConfiguration configuration) : base(configuration, RspecStrings.ResourceManager)
+        {
+        }
+
+        protected override void VisitAssignments(SyntaxNodeAnalysisContext context)
+        {
+            var identifier = (IdentifierNameSyntax)context.Node;
+            if (IsFileAccessPermissions(identifier, context.SemanticModel) && !identifier.IsPartOfBinaryNegationOrCondition())
+            {
+                context.ReportDiagnosticWhenActive(Diagnostic.Create(Rule, identifier.GetLocation()));
+            }
+        }
+
+        protected override void VisitInvocations(SyntaxNodeAnalysisContext context)
+        {
+            var invocation = (InvocationExpressionSyntax)context.Node;
+            if ((IsSetAccessRule(invocation, context.SemanticModel) || IsAddAccessRule(invocation, context.SemanticModel))
+                && (GetObjectCreation(invocation, context.SemanticModel) is { } objectCreation))
+            {
+                var invocationLocation = invocation.GetLocation();
+                var secondaryLocation = objectCreation.GetLocation();
+
+                var diagnostic = invocationLocation.GetLineSpan().StartLinePosition.Line == secondaryLocation.GetLineSpan().StartLinePosition.Line
+                    ? Diagnostic.Create(Rule, invocationLocation)
+                    : Diagnostic.Create(Rule, invocationLocation, additionalLocations: new[] {secondaryLocation});
+
+                context.ReportDiagnosticWhenActive(diagnostic);
+            }
+        }
+
+        private static bool IsFileAccessPermissions(SimpleNameSyntax identifierNameSyntax, SemanticModel semanticModel) =>
+            LooseFilePermissionsConfig.WeakFileAccessPermissions.Contains(identifierNameSyntax.Identifier.Text)
```

Comment:
Maybe identifierNameSyntax.Identifier could go into the SyntaxFacade class and if that is possible than the whole method could go to the base class.

Reply:
I'm not sure about this. In order to move `identifierNameSyntax.Identifier` there, we would need to use the base class `SyntaxNode` and do a cast in derived types. With this we will lose the type safety check which is now done at the compile time.

Reply:
Thanks for the tip with `Language.Syntax.NodeIdentifier(syntaxNode)`. Looks better now 👍